crates/apollo-mcp-server/src/server/states/running.rs

```diff
@@ -94,13 +94,18 @@ impl Running {
             operations.len(),
             serde_json::to_string_pretty(&operations)?
         );
-        *self.operations.lock().await = operations;
+        let mut operations_lock = self.operations.write().await;
+        *operations_lock = operations;
 
         // Update the schema itself
         *self.schema.lock().await = schema;
```

Comment:
Shouldn't we move this above `self.operations.write().await`?

I noticed that in the `update_schema()` function, we run `operations.write()` first and then `schema.lock()`. But in the `update_operations()` function below, we acquire the locks in the opposite order.

When one thread calls update_schema() and another thread calls update_operations(), this inconsistent lock ordering could potentially lead to a deadlock? 🤔